File: src/client/xnative/lion.xdroid/Setting/SettingPushActivity.cs
using Android.App;
using Android.Content.PM;
using Android.OS;
using Android.Support.V7.App;
using Android.Util;
using Android.Views;
using Android.Widget;
using Lion.XDroid;
using Java.Util;
using Lion.XDroid.Control;
using Lion.XDroid.Libs;
using Lion.XDroid.Type;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Lion.XDroid.Setting
{
    [Activity(Label = "SettingPushActivity", Theme = "@style/DefaultTheme", ScreenOrientation = ScreenOrientation.Portrait)]
    public class SettingPushActivity : AppCompatActivity
    {
        ListViewPushAdapter adapter;

        public SettingPushActivity()
        {
        }

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_setting_push);

            this.InitSize();
            this.InitEvent();
            this.InitData();
        }

        private void InitEvent()
        {
            var iv_back = FindViewById<ImageView>(Resource.Id.iv_back);
            iv_back.Touch += (s, e) =>
            {
                var iv_back1 = FindViewById<ImageView>(Resource.Id.iv_back1);
                if (e.Event.Action == MotionEventActions.Down)
                {
                    iv_back1.SetColorFilter(0xAA111111);
                }
                else if (e.Event.Action == MotionEventActions.Up)
                {
                    iv_back1.SetColorFilter(0x00000000);
                }

                e.Handled = false;
            };

            iv_back.Click += (s, e) =>
            {
                this.Finish();
            };
        }

        private void InitData()
        {
            var app_cache = new AppPreferences(this.ApplicationContext);
            var token = app_cache.UserTokenKey;
            {
                this.GetMessages(token);
                this.PushClear(token);
            }

            this.adapter = new ListViewPushAdapter(this, Resource.Layout.listview_push_custom, new List<PushMessage>());
            {
                var lv_push = FindViewById<ListView>(Resource.Id.lv_push);
                lv_push.Adapter = this.adapter;
            }
        }

        private void InitSize()
        {
            var iv_back = FindViewById<ImageView>(Resource.Id.iv_back);
            var iv_back1 = FindViewById<ImageView>(Resource.Id.iv_back1);
            var iv_line1 = FindViewById<ImageView>(Resource.Id.iv_line1);
            iv_back.LayoutParameters = AppCommon.SNG.ResizeDP(this, iv_back.LayoutParameters);
            iv_back1.LayoutParameters = AppCommon.SNG.ResizeDP(this, iv_back1.LayoutParameters);
            iv_line1.LayoutParameters = AppCommon.SNG.ResizeDP(this, iv_line1.LayoutParameters);
        }

        private void GetMessages(string user_token)
        {
            try
            {
                this.Loading(true);

                var _parameter = new Dictionary<string, string>();
                {
                    _parameter.Add("notify_time", AppCommon.SNG.GetDate(Calendar.Instance));
                }

                var api = new ApiAsnycTask(this, GetString(Resource.String.api_url) + "notify/GetMessages", user_token);
                api.Execute(_parameter);

                api.SendFinish += (s, e) =>
                {
                    var _result = JsonConvert.DeserializeObject<ApiResult<List<PushMessage>>>(e.Json);
                    if (_result != null)
                    {
                        if (_result.success == true)
                        {
                            this.adapter.Clear();
                            this.adapter.AddAll(_result.result);
                            this.adapter.NotifyDataSetChanged();

                            var tv_noting = FindViewById<TextView>(Resource.Id.tv_noting);
                            if (_result.result.Count == 0)
                                tv_noting.Visibility = ViewStates.Visible;
                            else
                                tv_noting.Visibility = ViewStates.Invisible;
                        }
                        else
                            AppDialog.SNG.Alert(this, _result.message);
                    }

                    this.Loading(false);
                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }

        private void PushClear(string user_token)
        {
            try
            {
                var _parameter = new Dictionary<string, string>();

                var api = new ApiAsnycTask(this, GetString(Resource.String.api_url) + "notify/Clear", user_token);
                api.Execute(_parameter);

                api.SendFinish += (s, e) =>
                {

                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }

        private void Loading(bool view)
        {
            var btn_loading = FindViewById<Button>(Resource.Id.btn_loading);
            var pb_loading = FindViewById<ProgressBar>(Resource.Id.pb_loading);
            if (view)
            {
                btn_loading.Visibility = ViewStates.Visible;
                pb_loading.Visibility = ViewStates.Visible;
            }
            else
            {
                btn_loading.Visibility = ViewStates.Invisible;
                pb_loading.Visibility = ViewStates.Invisible;
            }

        }
    }
}

File: src/client/xcode/Lion.Droid/app/src/main/java/com/odinsoftware/lottolion/setting/SettingPushActivity.java
package com.odinsoftware.lottolion.setting;

import android.graphics.PorterDuff;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.odinsoftware.lottolion.R;
import com.odinsoftware.lottolion.control.ListViewPushAdapter;
import com.odinsoftware.lottolion.service.ApiAsnycTask;
import com.odinsoftware.lottolion.service.LLAlert;
import com.odinsoftware.lottolion.service.LLCommon;
import com.odinsoftware.lottolion.service.LLSharedPreferences;
import com.odinsoftware.lottolion.type.ApiResult;
import com.odinsoftware.lottolion.type.PushMessage;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.HashMap;

public class SettingPushActivity extends AppCompatActivity {
    ArrayList<PushMessage> list = new ArrayList<>();
    ListViewPushAdapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting_push);

        InitSize();
        InitData();
        InitEvent();
    }

    private void InitEvent() {
        ImageView iv_back = (ImageView)findViewById(R.id.iv_back);

        iv_back.setOnTouchListener(listener_touch_back);
        iv_back.setOnClickListener(listener_back);
    }

    private void InitData() {
        String token = LLSharedPreferences.LL(getApplicationContext()).getSaveSession();

        if(LLCommon.Get().CheckExpired(token) == false){
            GetMessages(token);
            PushClear(token);
        }
        ListView lv_push = (ListView)findViewById(R.id.lv_push);
        adapter = new ListViewPushAdapter(this,R.layout.listview_push_custom, list);
        lv_push.setAdapter(adapter);
    }

    private void InitSize() {
        ImageView iv_back = (ImageView)findViewById(R.id.iv_back);
        ImageView iv_back1 = (ImageView)findViewById(R.id.iv_back1);
        ImageView iv_line1 = (ImageView)findViewById(R.id.iv_line1);

        iv_back.setLayoutParams(LLCommon.Get().ResizeDP(this,iv_back.getLayoutParams()));
        iv_back1.setLayoutParams(LLCommon.Get().ResizeDP(this,iv_back1.getLayoutParams()));
        iv_line1.setLayoutParams(LLCommon.Get().ResizeDP(this,iv_line1.getLayoutParams()));
    }
    private void GetMessages(String key){
        try{
            Loading(true);
            HashMap<String, String> parameter = new HashMap<>();
            parameter.put("notify_time",LLCommon.Get().GetDate(Calendar.getInstance()));
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"notify/GetMessages",key);
            api.execute(parameter);
            api.finish = new ApiAsnycTask.Finish() {
                @Override
                public void Finish(String json) {
                    Gson g = new Gson();
                    ApiResult<ArrayList<PushMessage>> result = g.fromJson(json, new TypeToken<ApiResult<ArrayList<PushMessage>>>(){}.getType());
                    if(result != null){
                        if(result.success){
                            list.clear();
                            list.addAll(result.result);
                            adapter.notifyDataSetChanged();
                            TextView tv_noting = (TextView)findViewById(R.id.tv_noting);
                            if(result.result.size() == 0)
                                tv_noting.setVisibility(View.VISIBLE);
                            else
                                tv_noting.setVisibility(View.INVISIBLE);

                        }
                        else{
                            LLAlert.Get(SettingPushActivity.this).Alert(result.message);
                        }
                    }
                    Loading(false);
                }
            };
        }
        catch (Exception ex){

        }
    }
    private void PushClear(String key){
        try{
            HashMap<String, String> parameter = new HashMap<>();
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"notify/Clear",key);
            api.execute(parameter);
        }
        catch (Exception ex){

        }
    }
    View.OnClickListener listener_back = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            finish();
        }
    };
    View.OnTouchListener listener_touch_back = new View.OnTouchListener() {
        @Override
        public boolean onTouch(View v, MotionEvent event) {
            // TODO Auto-generated method stub
            ImageView iv_back1 = (ImageView)findViewById(R.id.iv_back1);
            if(event.getAction() == MotionEvent.ACTION_DOWN){
                iv_back1.setColorFilter(0xaa111111, PorterDuff.Mode.SRC_OVER);
            }
            else if(event.getAction() == MotionEvent.ACTION_UP){
                iv_back1.setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
            }

            return false;
        }
    };
    private void Loading(boolean view){
        Button btn_loading = (Button)findViewById(R.id.btn_loading);
        ProgressBar pb_loading = (ProgressBar)findViewById(R.id.pb_loading);

        if(view){
            btn_loading.setVisibility(View.VISIBLE);
            pb_loading.setVisibility(View.VISIBLE);
        }
        else{
            btn_loading.setVisibility(View.INVISIBLE);
            pb_loading.setVisibility(View.INVISIBLE);
        }
    }
}
